firedrake/randomfunctiongen.py

```diff
@@ -106,7 +106,7 @@
 
 from firedrake.function import Function
 from pyop2.mpi import COMM_WORLD
-from ufl import FunctionSpace
+from ufl.functionspace import BaseFunctionSpace
```

Comment:
It's either `ufl.BaseFunctionSpace` or `firedrake.functionspaceimpl.WithGeometryBase`

Reply:
Please update the doc in the above. I realised that the current doc is slightly wrong. We should use `.WithGeometry` (and `.FiredrakeDualSpace`) instead of `.FunctionSpace`.

Reply:
We don't want the base class?

Reply:
Or should we test for the UFL base class in the code, and refer to concrete firedrake classes in the docstring?

Reply:
Done.